src/UBAFeeCalculator/UBAFeeSpokeCalculatorAnalog.ts

```diff
@@ -0,0 +1,229 @@
+        // and the trigger hurdle
+        resultant.netRunningBalanceAdjustment = resultant.netRunningBalanceAdjustment.add(
+          resultant.runningBalance.sub(upperBoundTriggerHurdle.target)
+        );
+        // Set the running balance to the trigger hurdle
+        resultant.runningBalance = upperBoundTriggerHurdle.target;
+      }
+
+      // If the lower trigger hurdle is surpassed, we need to return the trigger hurdle value
+      // as the running balance. This is because the trigger hurdle is the minimum value we'd like to
+      // organically shrink the running balance to. If the running balance is less than the trigger hurdle,
+      // we need to return the trigger hurdle as the running balance because at this point the dataworker
+      // will be triggered to rebalance the running balance.
+      else if (
+        lowerBoundTriggerHurdle !== undefined &&
+        resultant.runningBalance.lt(lowerBoundTriggerHurdle.threshold)
+      ) {
+        // Update the net running balance adjustment to reflect the difference between the running balance
+        // and the trigger hurdle
+        resultant.netRunningBalanceAdjustment = resultant.netRunningBalanceAdjustment.add(
+          lowerBoundTriggerHurdle.target.sub(resultant.runningBalance)
+        );
+        // Set the running balance to the trigger hurdle
+        resultant.runningBalance = lowerBoundTriggerHurdle.target;
+      }
+
+      return resultant;
+    },
+    {
+      runningBalance: lastValidatedRunningBalance ?? toBN(0),
+      incentiveBalance: lastValidatedIncentiveRunningBalance ?? toBN(0),
+      netRunningBalanceAdjustment: toBN(0),
+    }
+  );
+  // Return the result
+  return historicalResult;
+}
+
+export function getEventFee(
+  amount: BigNumber,
+  flowType: "inflow" | "outflow",
+  previousFlows: UbaFlow[],
+  lastValidatedRunningBalance: BigNumber,
+  lastValidatedIncentiveRunningBalance: BigNumber,
+  chainId: number,
+  tokenSymbol: string,
+  config: UBAConfig
+): UBAFlowFee {
+  // The rough psuedocode for this function is as follows:
+  // We'll need two inflow/outflow curves
+  // We need to determine which flow curve to use based on the flow type
+  // Compute first balancing fee <- f(x, x+amnt)
+  // Compute second balance fee (oportunity cost) <- g(x+amnt, x)
+  // Incentive Fee (LP Fee Component): first balancing fee - second balance fee
+  // Return (LP Fee + Balancing Fee)
+  // #############################################
+
+  // We'll need to now compute the concept of the running balance of the spoke
+  const { runningBalance } = calculateHistoricalRunningBalance(
+    previousFlows,
+    lastValidatedRunningBalance,
+    lastValidatedIncentiveRunningBalance,
+    chainId,
+    tokenSymbol,
+    config
+  );
+
+  // We first need to resolve the inflow/outflow curves for the deposit and refund spoke
+  const flowCurve = config.getBalancingFeeTuples(chainId);
```

Comment:
Same as this: https://github.com/across-protocol/sdk-v2/pull/266/files#r1253650456

Reply:
Resolved offline 👀